diff --git a/auto-post-thumbnail/includes/class-wapt-base.php b/auto-post-thumbnail/includes/class-wapt-base.php
index 4c8546f8..86a9d58b 100644
--- a/auto-post-thumbnail/includes/class-wapt-base.php
+++ b/auto-post-thumbnail/includes/class-wapt-base.php
@@ -588,9 +588,9 @@ class AutoPostThumbnails {
 	public function publish_post( $post_id ) {
 		global $wpdb;
 
-		if ( ! is_single( $post_id ) ) {
-			return 0;
-		}
+		//if ( ! is_single( $post_id ) ) {
+		//	return 0;
+		//}
 
 		// First check whether Post Thumbnail is already set for this post.
 		$_thumbnail_id = get_post_meta( $post_id, '_thumbnail_id', true );
