diff --git a/basic-comment-quicktags/quicktags.js b/basic-comment-quicktags/quicktags.js
index e3e3f6a..d760990 100644
--- a/basic-comment-quicktags/quicktags.js
+++ b/basic-comment-quicktags/quicktags.js
@@ -16,7 +16,6 @@
       id: "posttext",
       buttons: "link,em,strong"
     });
-    QTags.addButton('quote','quote','<blockquote>','</blockquote>','quote','');
-    QTags.addButton('image','img','[img]','[/img]','I','');
+    QTags.addButton('image','resim','[img]','[/img]','I','');
 //   });
 // })(jQuery);
\ No newline at end of file
