diff --git a/bknewsticker/ticker-color.php b/bknewsticker/ticker-color.php
index ad552ce..0b1ca1b 100644
--- a/bknewsticker/ticker-color.php
+++ b/bknewsticker/ticker-color.php
@@ -20,7 +20,7 @@ background: <?php $color_background = get_option('ticker_background_posts'); ech
 }
 #bknews-title h3{
 	 color: <?php $color_title = get_option('ticker_color_title'); echo $color_title;?>;
-	 border-bottom: 40px solid <?php $background_title = get_option('ticker_background_title'); echo $background_title;?>;
+	 border-bottom: 25px solid <?php $background_title = get_option('ticker_background_title'); echo $background_title;?>;
 	 font-size: <?php $size_exerpt_text = get_option('bke_textall_size'); echo $size_exerpt_text;?>px;
 }
 #allticker{
@@ -30,4 +30,4 @@ background: <?php $color_background = get_option('ticker_background_posts'); ech
 	color: <?php $color_exerpt = get_option('ticker_color_link'); echo $color_exerpt;?>;
 	font-size: <?php $size_exerpt_text = get_option('bke_textall_size'); echo $size_exerpt_text;?>px;
 }
-</style>	
\ No newline at end of file
+</style>	
diff --git a/bknewsticker/ticker.css b/bknewsticker/ticker.css
index 21440da..1f144c7 100644
--- a/bknewsticker/ticker.css
+++ b/bknewsticker/ticker.css
@@ -5,6 +5,10 @@
 	border-bottom: 0px solid #ccc;
 }
 
+#allticker > ul > li {
+    padding-top: 2px;
+}
+
 #bknewsticker{
 	background: #F3F3F3;
 	margin: 0px;
@@ -16,7 +20,7 @@
 	background: #5A5A5A;
 	color: #ccc;
 	font-size: 10px;
-	line-height: 39px;
+	line-height: 24px;
 	margin-left: 3px;
 }
 .tickerLink a, .tickermore a{
@@ -27,14 +31,14 @@
 }
 .tickerText{
 	color: #5A5A5A;
-	line-height: 40px;
+	line-height: 25px;
 	font-size: 12px;
 
 }
 #bknews-title{
 	float: left;
 	width: auto;
-	height: 40px;
+	height: 25px;
 	background: transparent;
 	
 }
@@ -43,11 +47,11 @@
 	font-size: 14px;
 	margin: 0px;
 	text-align: center;
-	text-transform: uppercase;
+	//text-transform: uppercase;
 	text-decoration: none;
-    border-right: 20px solid transparent;
+    border-right: 25px solid transparent;
     height: 0;
-    line-height: 40px;
+    line-height: 25px;
     padding-left: 5px;
 }
 
@@ -56,4 +60,4 @@
 	}
 @media only screen and (max-width: 320px) {
 	#bknews-title{ display: none!important;
-	}
\ No newline at end of file
+	}
diff --git a/bknewsticker/ticker.php b/bknewsticker/ticker.php
index 139c966..b809726 100644
--- a/bknewsticker/ticker.php
+++ b/bknewsticker/ticker.php
@@ -8,7 +8,8 @@ Author URI: http://www.bekero.com
 Version: 1.0.3
 */
 define('TICKER_VERSION', '1.0.3');
-define('TICKER_MAX_INT', defined('PHP_INT_MAX') ? PHP_INT_MAX : 32767);
+// define('TICKER_MAX_INT', defined('PHP_INT_MAX') ? PHP_INT_MAX : 32767);
+define('TICKER_MAX_INT', 100);
 define('PHPREQ',5);
 $phpver=phpversion();$phpmaj=$phpver[0];
 function ticker_add_pages() {
@@ -63,7 +64,7 @@ function insert_bknewsticker(){
 		jQuery('#bknewsticker').cycle({ 
 			speed: <?php echo $tickerspeed; ?>000,
 			timeout: <?php echo $tickertimeout; ?>000,
-			height: 40,
+			height: 25,
 			fx: '<?php echo $tickeranimation; ?>',
 			pause: 1,
 			containerResize: 1
@@ -183,6 +184,9 @@ function ticker_recent_comments($src_count, $src_length) {
 }
 
 function ticker_get_posts($type, $cat_filter, $n, $post_list=null){
+	// We do not want to fetch 2147483647 posts
+	$get_posts_category_param = count($cat_filter) > 0 ? $cat_filter[0] : '';
+
 	switch($type){
 		case 'popular':
 		$days = get_option('ticker_popular_days');
@@ -199,7 +203,8 @@ function ticker_get_posts($type, $cat_filter, $n, $post_list=null){
 		case 'recent':
 		$posts = get_posts(
 			array(
-				'numberposts' => TICKER_MAX_INT, 
+				'numberposts' => $n ? $n : TICKER_MAX_INT,
+				'category' => $get_posts_category_param,
 				'orderby' => 'post_date',
 				'suppress_filters' => 0,
 				)
@@ -208,7 +213,8 @@ function ticker_get_posts($type, $cat_filter, $n, $post_list=null){
 		case 'commented':
 		$posts = get_posts(
 			array(
-				'numberposts' => TICKER_MAX_INT,
+				'numberposts' => $n ? $n : TICKER_MAX_INT,
+				'category' => $get_posts_category_param,
 				'orderby' => 'comment_count',
 				'suppress_filters' => 0,
 				)
@@ -217,7 +223,8 @@ function ticker_get_posts($type, $cat_filter, $n, $post_list=null){
 		case 'userspecified':
 		$posts_tmp = get_posts(
 			array(
-				'numberposts' => TICKER_MAX_INT,
+				'numberposts' => $n ? $n : TICKER_MAX_INT,
+				'category' => $get_posts_category_param,
 				'include' => $post_list,
 				'suppress_filters' => 0,
 				)
@@ -861,4 +868,4 @@ function ticker_plugin_action_links( $links, $file ) {
 	}
 	return $links;
 }
-?>
\ No newline at end of file
+?>
