diff --git a/cubepm/cpm_admin_settings.php b/cubepm/cpm_admin_settings.php
index ad86ad2..5a027da 100644
--- a/cubepm/cpm_admin_settings.php
+++ b/cubepm/cpm_admin_settings.php
@@ -6,6 +6,18 @@
  */
 
 
+function cpm_get_user_role_names() {
+  global $wp_roles;
+
+  if (!isset($wp_roles)) {
+    echo "not set";
+    $wp_roles = new WP_Roles();
+  }
+
+  return $wp_roles->get_names();
+}
+
+
 /**
  * CubePM Admin Settings Page
  * 
@@ -56,6 +68,7 @@ function cpm_admin_settings(){
 	$cpm_email_body = htmlspecialchars(get_option('cpm_email_body'));
 	$cpm_permission_newtopic = (array) get_option('cpm_permission_newtopic');
 ?>
+
 <div class="wrap"> 
 	<div id="icon-options-general" class="icon32"><br /></div> 
 <h2>CubePM - <?php _e('Settings', 'cubepm'); ?></h2> 
@@ -71,11 +84,21 @@ function cpm_admin_settings(){
 <th scope="row"><?php _e('Allowed to start PM topics', 'cubepm'); ?></th> 
 <td><fieldset>
 <p>
+<!--
 <label><input name="cpm_permission_newtopic[]"  type="checkbox" value="administrator" <?php echo in_array('administrator', $cpm_permission_newtopic) ? 'checked="checked"' : ''  ; ?> /> <?php _e('Administrators', 'cubepm'); ?></label><br />
 <label><input name="cpm_permission_newtopic[]"  type="checkbox" value="editor" <?php echo in_array('editor', $cpm_permission_newtopic) ? 'checked="checked"' : ''  ; ?> /> <?php _e('Editors', 'cubepm'); ?></label><br />
 <label><input name="cpm_permission_newtopic[]"  type="checkbox" value="author" <?php echo in_array('author', $cpm_permission_newtopic) ? 'checked="checked"' : ''  ; ?> /> <?php _e('Authors', 'cubepm'); ?></label><br />
 <label><input name="cpm_permission_newtopic[]"  type="checkbox" value="contributor" <?php echo in_array('contributor', $cpm_permission_newtopic) ? 'checked="checked"' : ''  ; ?> /> <?php _e('Contributors', 'cubepm'); ?></label><br />
 <label><input name="cpm_permission_newtopic[]"  type="checkbox" value="subscriber" <?php echo in_array('subscriber', $cpm_permission_newtopic) ? 'checked="checked"' : ''  ; ?> /> <?php _e('Subscribers', 'cubepm'); ?></label>
+-->
+<?php
+$current_roles = cpm_get_user_role_names();
+
+foreach ($current_roles as $role_name => $role_nice_name) {
+  $checked_attr = in_array($role_name, $cpm_permission_newtopic) ? 'checked="checked"' : '';
+  echo('<label><input name="cpm_permission_newtopic[]" type="checkbox" value="' . $role_name . '" ' . $checked_attr . ' /> ' . $role_nice_name . '</label><br>');
+}
+?>
 </p> 
 </fieldset></td> 
 </tr>
@@ -135,4 +158,4 @@ function cpm_admin_settings(){
  
 </div>
 <?php
-}
\ No newline at end of file
+}
diff --git a/cubepm/css/cubepm.css b/cubepm/css/cubepm.css
index 2bc3f54..fa61e0e 100644
--- a/cubepm/css/cubepm.css
+++ b/cubepm/css/cubepm.css
@@ -180,6 +180,11 @@
 	padding: 0;
 }
 
+.cpm-thread {
+  table-layout: fixed;
+  word-wrap: break-word;
+}
+
 .cubepm .cpm-thread-reply,.cubepm .cpm-thread-reply td {
 	background: none !important;
 	padding: 0 !important;
@@ -271,4 +276,4 @@
 .ac_over {
 	background-color: #ddd;
 	color: #000;
-}
\ No newline at end of file
+}
