diff --git a/editable-comments/editable-comments.php b/editable-comments/editable-comments.php
index 25c9888e..3ff02e1f 100644
--- a/editable-comments/editable-comments.php
+++ b/editable-comments/editable-comments.php
@@ -9,7 +9,7 @@ Author URI: http://julienappert.com
 */
 class WPEditableComments{
 
-	function WPEditableComments(){$this->__construct();}
+	// function WPEditableComments(){$this->__construct();}
 		
 	function __construct(){
 		add_action('init', array(&$this, 'init'));
@@ -140,10 +140,17 @@ class WPEditableComments{
 			
 	
 	function qualifications($comment,$type){
-		global $post;
-		if(current_user_can( 'edit_post', $post->ID ) || ($_SERVER['REMOTE_ADDR'] == $comment->comment_author_IP && WPEditableComments::dateValid($comment->comment_date_gmt,$type)))
+		global $post, $current_user;
+			
+		if (current_user_can('subscriber')) {
+			return false;
+		}
+		
+		if (($comment->user_id == $current_user->ID) || current_user_can('moderate_comments')) {
 			return true;
-		return false;	
+		}
+			
+		return false;
 	}
 	
 	// for backward compatibility
@@ -183,7 +190,7 @@ class WPEditableComments{
 		if($date >= time() - $minutes * 60)
 			return true;
 
-			return false;	
+			return true;	
 		
 	}
 	
