diff --git a/editable-comments/editable-comments.php b/editable-comments/editable-comments.php
index d9210a3..171b2a6 100644
--- a/editable-comments/editable-comments.php
+++ b/editable-comments/editable-comments.php
@@ -140,8 +140,9 @@ class WPEditableComments{
 			
 	
 	function qualifications($comment,$type){
-		global $post;
-		if(current_user_can( 'edit_post', $post->ID ) || ($_SERVER['REMOTE_ADDR'] == $comment->comment_author_IP && WPEditableComments::dateValid($comment->comment_date_gmt,$type)))
+		global $post, $current_user;
+    
+        if (($comment->user_id == $current_user->ID) || (current_user_can('edit_others_posts')))
 			return true;
 		return false;	
 	}
@@ -183,7 +184,7 @@ class WPEditableComments{
 		if($date >= time() - $minutes * 60)
 			return true;
 
-			return false;	
+			return true;	
 		
 	}
 	
