diff --git a/fluxophy/css/fluxophy.css b/fluxophy/css/fluxophy.css
index e0259c8..ba77f88 100644
--- a/fluxophy/css/fluxophy.css
+++ b/fluxophy/css/fluxophy.css
@@ -31,9 +31,8 @@ div.fluxophy_entry_data {
 
 .fluxophy_entry_header img {
   margin: 4px 4px 0 4px;
-  border: 1px solid #D2D2D2;
   border-radius: 4px;
-  width: 48px;
+  width: 32px;
 }
 
 .fluxophy_entry_header a {
@@ -116,4 +115,18 @@ div.fluxophy_entry_data {
   fill: #fff;
 }
 
+.fluxophy_22dakika {
+  display: inline-block;
+  font-weight: bold;
+  line-height: 15px; 
+  position: relative;
+  top: 5px;
+}
 
+.fluxophy_22dakika a:hover{
+  color: black;
+}
+
+.fluxophy_22dakika a:hover span {
+  text-decoration: none;
+}
diff --git a/fluxophy/includes/class-wp-fp-widget.php b/fluxophy/includes/class-wp-fp-widget.php
index 326ca6c..2907652 100644
--- a/fluxophy/includes/class-wp-fp-widget.php
+++ b/fluxophy/includes/class-wp-fp-widget.php
@@ -145,7 +145,7 @@ function fluxophy_fetch_data_fb($source_url, $display_count, $picture_url) {
         if (isset($picture_url)) {
           $output .= '<img src="' . $picture_url . '"></img>';
         }
-
+        $output .= "<div class='fluxophy_22dakika'><a href='http://www.facebook.com/22dakika.org'>22dakika<br/><span style='font-weight:normal'>/22dakika.org</span></a></div>";
         $output .= '<a href="' . $received->data[$i]->link . '" title="' . date_i18n('G:i - d F Y', strtotime($received->data[$i]->updated_time)) . '" target="_blank">';
         $output .= date_i18n('d M', strtotime($received->data[$i]->updated_time));
         $output .= '</a>';
