diff --git a/login-logout/login-logout.php b/login-logout/login-logout.php
index f7328b2..4c0e5e1 100644
--- a/login-logout/login-logout.php
+++ b/login-logout/login-logout.php
@@ -17,6 +17,7 @@ class WP_Widget_Login_Logout extends WP_Widget {
 	}
 	
 	public function widget( $args, $instance ) { // outputs the content of the widget
+    global $wpdb;
 		extract($args);
 		//$title = apply_filters('widget_title', empty($instance['title']) ? __('Login-Logout', 'login-logout') : $instance['title'], $instance, $this->id_base);
 		$title = apply_filters('widget_title', $instance['title']);
@@ -56,7 +57,7 @@ class WP_Widget_Login_Logout extends WP_Widget {
 			$wrap_after = '</p>';
 			$item_before = '<span class='; // class will be added and the tag is not closed
 			$item_after = '</span>';
-			$split_char = ' | ';
+      $split_char = '';
 		} else {
 			$wrap_before = '<ul class="wrap_login_logout">';
 			$wrap_after = '</ul>';
@@ -70,32 +71,52 @@ class WP_Widget_Login_Logout extends WP_Widget {
 			if ( is_user_logged_in() ){
 				$current_user = wp_get_current_user();
 				$username = $current_user->display_name;
-				$username_link = apply_filters( 'login_logout_username_link', '<a href="'.admin_url('profile.php').'">'.$username.'</a>', $current_user );
+				$username_link = '<a class="kabartmatozu" href="'.get_author_posts_url($current_user->ID).'">'.$username.'</a>';
 				$welcome_text_new = str_replace('[username]', $username_link, $welcome_text);
-				echo $item_before.'"item_welcome">'.$welcome_text_new.$item_after.$split_char;
+				echo $item_before.'"item_welcome">'.$welcome_text_new.$item_after;
 			}
 		}
 		echo $item_before;
 		//wp_loginout( $redirect_to_self );
 		if ( ! is_user_logged_in() ){
 			echo '"item_login">';
-			echo '<a href="'.esc_url( wp_login_url( $login_redirect_to ) ).'">'.$login_text.'</a>';
+			echo '<a class="kabartmatozu kabartmatozu_color_red" href="'.esc_url( wp_login_url( $login_redirect_to ) ).'">'.$login_text.'</a>';
 		} else {
 			echo '"item_logout">';
-			echo '<a href="'.esc_url( wp_logout_url( $logout_redirect_to ) ).'">'.$logout_text.'</a>';
+			$current_user = wp_get_current_user();
+			$postane_threads = $wpdb->prefix . 'postane_threads';
+			$postane_user_thread = $wpdb->prefix . 'postane_user_thread';
+			$current_uid = $current_user->ID;
+			$rows = $wpdb->get_results("SELECT count(*) as unread_msg_count from $postane_threads INNER JOIN $postane_user_thread ON $postane_user_thread.thread_id=$postane_threads.id where $postane_user_thread.user_id=$current_uid AND $postane_user_thread.last_read_time < $postane_threads.last_message_time", ARRAY_A);
+
+			$unread_message_count = 0;			
+			foreach ( $rows as $row ) {
+			    $unread_message_count = $row['unread_msg_count'];			
+			}
+			
+			$unread_str = '';
+			$unread_class = '';
+			if ($unread_message_count > 0){
+				// $unread_str = ' <sup class="counter"><blink>('.$unread_message_count.')</blink></sup>';
+				$unread_str = ' <span class="blink">(' . $unread_message_count . ')</span>';
+				$unread_class = 'kabartmatozu_color_red';
+			}			
+			
+			echo '<a class="kabartmatozu ' . $unread_class . '" href="/postane">postane' . $unread_str . '</a>' .
+				' <a class="kabartmatozu" href="' . esc_url( wp_logout_url( $logout_redirect_to ) ) . '">' . $logout_text . '</a>';
 		}
 		echo $item_after;
 		//wp_register();
 		if( $register_link ){ // register link
 			if ( ! is_user_logged_in() ) {
 				if ( get_option('users_can_register') ){
-					echo $split_char.$item_before.'"item_register">'.'<a href="'.site_url('wp-login.php?action=register', 'login').'">'.$register_text.'</a>'.$item_after;
+					echo $split_char.$item_before.'"item_register">'.'<a class="kabartmatozu kabartmatozu_color_red" href="'.site_url('wp-login.php?action=register', 'login').'">'.$register_text.'</a>'.$item_after;
 				}
 			}
 		}
 		if( $admin_link ){ // admin link
 			if ( is_user_logged_in() ) {
-				echo $split_char.$item_before.'"item_admin">'.'<a href="'.admin_url().'">'.$admin_text.'</a>'.$item_after;
+				echo $split_char.$item_before.'"item_admin">'.'<a class="kabartmatozu" href="'.admin_url().'">'.$admin_text.'</a>'.$item_after;
 			}
 		}
 		
@@ -258,4 +279,4 @@ if ( ! function_exists( 'login_logout_unqprfx_plugin_meta' ) ) :
 		return $links;
 	}
 	add_filter( 'plugin_row_meta', 'login_logout_unqprfx_plugin_meta', 10, 2 );
-endif; // end of login_logout_unqprfx_plugin_meta()
\ No newline at end of file
+endif; // end of login_logout_unqprfx_plugin_meta()
