diff --git a/permalink-editor/index.php b/permalink-editor/index.php
index f1b1d487..4197dcca 100644
--- a/permalink-editor/index.php
+++ b/permalink-editor/index.php
@@ -69,7 +69,8 @@ class Permalink_Editor
 	 * use priority 11 here to try and catch anything that may have been added
 	 * on or before the default priority of 10 (such as custom post types).
 	 */
-	function Permalink_Editor()
+	// function Permalink_Editor()
+	function __construct()
 	{
 		add_action( 'init', array( &$this, 'init' ), 11 );
 		add_action( 'admin_init', array( &$this, 'admin_init' ) );
@@ -817,4 +818,4 @@ class Permalink_Editor
 
 $PermalinkEditor = new Permalink_Editor();
 
-?>
\ No newline at end of file
+?>
