diff --git a/simple-local-avatars/simple-local-avatars.php b/simple-local-avatars/simple-local-avatars.php
index bf0cd5d7..6e47bfd3 100644
--- a/simple-local-avatars/simple-local-avatars.php
+++ b/simple-local-avatars/simple-local-avatars.php
@@ -63,12 +63,15 @@ class Simple_Local_Avatars {
 	 * @return string <img> tag for the user's avatar
 	 */
 	public function get_avatar( $avatar = '', $id_or_email = '', $size = 96, $default = '', $alt = '' ) {
+		global $wpdb;
 		if ( is_numeric( $id_or_email ) )
 			$user_id = (int) $id_or_email;
 		elseif ( is_string( $id_or_email ) && ( $user = get_user_by( 'email', $id_or_email ) ) )
 			$user_id = $user->ID;
 		elseif ( is_object( $id_or_email ) && ! empty( $id_or_email->user_id ) )
 			$user_id = (int) $id_or_email->user_id;
+		else
+			$user_id = $wpdb->get_var("SELECT user_id FROM wp_comments WHERE comment_author_email = '" . $id_or_email . "' LIMIT 1");
 		
 		if ( empty( $user_id ) )
 			return $avatar;
@@ -141,6 +144,11 @@ class Simple_Local_Avatars {
 		
 		$author_class = is_author( $user_id ) ? ' current-author' : '' ;
 		$avatar = "<img alt='" . esc_attr( $alt ) . "' src='" . esc_url( $local_avatars[$size] ) . "' class='avatar avatar-{$size}{$author_class} photo' height='{$size}' width='{$size}' />";
+		$avatar_url = esc_url($local_avatars[$size]);
+		if (!empty($avatar_url)) {
+			$avatar_url = set_url_scheme($avatar_url);
+		}
+		$avatar = "<img alt='" . esc_attr( $alt ) . "' src='" . $avatar_url  . "' class='avatar avatar-{$size}{$author_class} photo' height='{$size}' width='{$size}' />";
 		
 		return apply_filters( 'simple_local_avatar', $avatar );
 	}
