diff --git a/simple-local-avatars/includes/class-simple-local-avatars.php b/simple-local-avatars/includes/class-simple-local-avatars.php
index 2148b696..d12d0d0b 100644
--- a/simple-local-avatars/includes/class-simple-local-avatars.php
+++ b/simple-local-avatars/includes/class-simple-local-avatars.php
@@ -301,6 +301,7 @@ class Simple_Local_Avatars {
 	 * @return int|false
 	 */
 	public function get_user_id( $id_or_email ) {
+		global $wpdb;
 		$user_id = false;
 
 		if ( is_numeric( $id_or_email ) ) {
@@ -314,6 +315,8 @@ class Simple_Local_Avatars {
 		} elseif ( is_string( $id_or_email ) ) {
 			$user    = get_user_by( 'email', $id_or_email );
 			$user_id = $user ? $user->ID : '';
+		} else {
+			$user_id = $wpdb->get_var("SELECT user_id FROM wp_comments WHERE comment_author_email = '" . $id_or_email . "' LIMIT 1");
 		}
 
 		return $user_id;
