diff --git a/spoiler-block/css/style.css b/spoiler-block/css/style.css
index bab07e3..f8c104b 100644
--- a/spoiler-block/css/style.css
+++ b/spoiler-block/css/style.css
@@ -1,17 +1,18 @@
-/* just code */
-.spoiler{
-	background:#FF0000;
-	border-radius:5px;
-	padding:5px;
-	font-weight:bold;
-	color:#FFFFFF;
-	cursor:pointer;
-}
-.spoiler-open{
-	background:#FFDEDE;
-	border-radius:5px;
-	padding:5px;
-}
-.spoiler > .hidden-content{
-	display: none;
+/* just code */
+.spoiler{
+	background:#000000;
+	border-radius:5px;
+	padding: 4px 5px 3px 5px;
+	margin: 0 5px;
+	font-weight:bold;
+	color:#FFFFFF;
+	cursor:pointer;
+}
+.spoiler-open{
+	background:#FFDEDE;
+	border-radius:5px;
+	padding:5px;
+}
+.spoiler > .hidden-content{
+	display: none;
 }
\ No newline at end of file
diff --git a/spoiler-block/spoiler_block.php b/spoiler-block/spoiler_block.php
index c5f33bb..b93eed7 100644
--- a/spoiler-block/spoiler_block.php
+++ b/spoiler-block/spoiler_block.php
@@ -53,13 +53,14 @@ if (!defined('SPOILERBLOCK_CONFIG_PAGE')) {
   define('SPOILERBLOCK_CONFIG_PAGE', "spoiler-block-config");
 }
 
-add_action('wp_print_styles', 'add_sb_style');
-add_action('wp_enqueue_scripts', 'add_sb_scripts');
+// add_action('wp_print_styles', 'add_sb_style');
+// add_action('wp_enqueue_scripts', 'add_sb_scripts');
 
 load_plugin_textdomain( SPOILERBLOCK_TEXTDOMAIN, false, "/spoiler-block/languages" );
 
 /* enfileirando os marotos */
 
+/*
 function add_sb_style() {
     $myStyleUrl = SPOILERBLOCK_PLUGIN_URL . '/css/style.css';
 
@@ -74,7 +75,16 @@ function add_sb_scripts() {
     	array('scriptaculous'),
     	'1.0', true );
 }
+ */
+
+function spoiler_block_add_assets() {
+  wp_register_style('spoiler_block_css', plugins_url('css/style.css', __FILE__));
+  wp_enqueue_style('spoiler_block_css');
 
+  wp_register_script('spoiler_block_js', plugins_url('js/scripts.js', __FILE__), array('jquery'));
+  wp_enqueue_script('spoiler_block_js');
+}
+add_action('wp_enqueue_scripts', 'spoiler_block_add_assets');
 
 /*
 	Fazendo o nego funfar no admin =D
@@ -92,7 +102,8 @@ function spoiler_add_button($buttons)
  
 function spoiler_register($plugin_array){
 
-    $url = SPOILERBLOCK_PLUGIN_URL . '/js/spoiler_plugin/spoiler_mce.js';
+    // $url = SPOILERBLOCK_PLUGIN_URL . '/js/spoiler_plugin/spoiler_mce.js';
+    $url = plugins_url('js/spoiler_plugin/spoiler_mce.js', __FILE__);
  
     $plugin_array["spoiler"] = $url;
     return $plugin_array;
@@ -106,7 +117,8 @@ function spoiler_editor_css($url) {
 	if ( !empty($url) )
 		$url .= ',';
 
-	$url .= SPOILERBLOCK_PLUGIN_URL . '/css/spoiler_admin_style.css';
+	// $url .= SPOILERBLOCK_PLUGIN_URL . '/css/spoiler_admin_style.css';
+	$url .= plugins_url('css/spoiler_admin_style.css', __FILE__);
 
   return $url;
 }
@@ -134,4 +146,16 @@ function spoiler_selected_message(){
 }
 
 add_action("wp_head","spoiler_selected_message");
-?>
\ No newline at end of file
+
+function ispiyon_shortcode_button() {
+  if (wp_script_is('quicktags')) {
+    ?>
+    <script type="text/javascript">
+      QTags.addButton('ispiyon_qtag','ispiyon','<span class="spoiler"><span class="hidden-content">','</span></span>');
+    </script>
+    <?php
+  }
+}
+add_action('admin_print_footer_scripts', 'ispiyon_shortcode_button');
+
+?>
