diff --git a/w3-total-cache/ObjectCache_WpObjectCache_Regular.php b/w3-total-cache/ObjectCache_WpObjectCache_Regular.php
index 4e544b1..a670b3c 100755
--- a/w3-total-cache/ObjectCache_WpObjectCache_Regular.php
+++ b/w3-total-cache/ObjectCache_WpObjectCache_Regular.php
@@ -153,7 +153,7 @@ class ObjectCache_WpObjectCache_Regular {
 			!in_array( $group, $this->nonpersistent_groups ) &&
 			$this->_check_can_cache_runtime( $group ) ) {
 			$cache = $this->_get_cache( null, $group );
-			$v = $cache->get( $key );
+			$v = $cache->get( $key, $group );
 
 			/* for debugging
 				$a = $cache->_get_with_old_raw( $key );
@@ -277,7 +277,7 @@ class ObjectCache_WpObjectCache_Regular {
 
 			$v = array( 'content' => $data );
 			$ext_return = $cache->set( $key, $v,
-				( $expire ? $expire : $this->_lifetime ) );
+				( $expire ? $expire : $this->_lifetime ), $group );
 			$return = $ext_return;
 		}
 
@@ -319,7 +319,7 @@ class ObjectCache_WpObjectCache_Regular {
 
 		if ( $this->_caching && !in_array( $group, $this->nonpersistent_groups ) ) {
 			$cache = $this->_get_cache( null, $group );
-			$return = $cache->delete( $key );
+			$return = $cache->delete( $key, $group );
 		}
 
 		if ( $this->_is_transient_group( $group ) &&
