diff --git a/wti-like-post/css/wti_like_post.css b/wti-like-post/css/wti_like_post.css
index 7cce199..a1e707a 100644
--- a/wti-like-post/css/wti_like_post.css
+++ b/wti-like-post/css/wti_like_post.css
@@ -2,6 +2,7 @@
 	float: left;
 	width: 100%;
 	height: 100%;
+	position: relative;
 }
 
 #watch_action img {
@@ -44,12 +45,28 @@
 .unlc {
 	color: red;		
 }
+.wti_like_h3{
+	margin-bottom: 30px;
+	border-bottom: 1px dotted #d2d2d2;
+	font: normal 24px/1.2em museo, 
+	helvetica,Arial;margin-top: 10px;
+}
+.wti_like_message{
+	color: gray;
+	padding: 0 7px 0 7px;
+	font-size: 16px;
+	line-height: 30px;
+	position: absolute;
+	top: 15px;
+	left: 8.3%
+}
 
 .status {
 	color: gray;
 	padding: 0 7px 0 7px;
 	font-size: 14px;
-	line-height: 18px;	
+	line-height: 18px;
+	/*margin: -20px 50px;*/
 }
 .status .status_msg {
 	width: auto;
diff --git a/wti-like-post/js/wti_like_post.js b/wti-like-post/js/wti_like_post.js
index 5619eb5..56d7176 100644
--- a/wti-like-post/js/wti_like_post.js
+++ b/wti-like-post/js/wti_like_post.js
@@ -2,7 +2,14 @@ jQuery(document).ready(function(){
      jQuery(".like img, .unlike img").click(function(){
           var task = jQuery(this).attr("rel");
           var post_id = jQuery(this).attr("id");
-          
+
+          var yildiz = 'dyildiz.png';
+          var current_yildiz = jQuery(this).attr('src');
+
+          if (current_yildiz.endsWith(yildiz)) {
+              yildiz = 'byildiz.png';
+          }
+
           if(task == "like")
           {
                post_id = post_id.replace("like-", "");
@@ -19,6 +26,8 @@ jQuery(document).ready(function(){
                url: blog_url + "/wp-content/plugins/wti-like-post/wti_like.php",
                data: "post_id=" + post_id + "&task=" + task + "&num=" + Math.random(),
                success: function(data){
+                    // this should be img#like- and img#unlike- but we are using this plugin wrong...
+                    jQuery('img#like-' + post_id).attr('src', blog_url + '/wp-content/plugins/wti-like-post/images/' + yildiz);
                     jQuery("#lc-" + post_id).html(data.like);
                     jQuery("#unlc-" + post_id).html(data.unlike);
                     jQuery("#status-" + post_id).removeClass("loading-img").empty().html(data.msg);
@@ -26,4 +35,4 @@ jQuery(document).ready(function(){
                dataType: "json"
           });
      });
-});
\ No newline at end of file
+});
diff --git a/wti-like-post/wti_like.php b/wti-like-post/wti_like.php
index f489361..bbcd7fe 100644
--- a/wti-like-post/wti_like.php
+++ b/wti-like-post/wti_like.php
@@ -7,6 +7,7 @@ global $wpdb;
 $post_id = (int)$_POST['post_id'];
 $task = $_POST['task'];
 $ip = $_SERVER['REMOTE_ADDR'];
+$cur_user = wp_get_current_user();
 
 //get setting data
 $is_logged_in = is_user_logged_in();
@@ -18,14 +19,15 @@ if($login_required && !$is_logged_in) {
 	$error = 1;
 	$msg = get_option('wti_like_post_login_message');
 } else {
-	$has_already_voted = HasWtiAlreadyVoted($post_id, $ip);
+	$has_already_voted = HasWtiAlreadyVoted($post_id, $cur_user->ID);
 	$voting_period = get_option('wti_like_post_voting_period');
 	$datetime_now = date('Y-m-d H:i:s');
-	
 	if("once" == $voting_period && $has_already_voted) {
+    $query = 'DELETE FROM wp_wti_like_post WHERE post_id = ' . $post_id . ' AND user_id = ' . $cur_user->ID;
+    $wpdb->query($query);
 		//user can vote only once and has already voted.
 		$error = 1;
-		$msg = get_option('wti_like_post_voted_message');
+		$msg = get_option('wti_like_post_allowed_posts');
 	} elseif(0 == $voting_period) {
 		//user can vote as many times as he want
 		$can_vote = true;
@@ -69,7 +71,8 @@ if($can_vote) {
 			$query .= "post_id = '" . $post_id . "', ";
 			$query .= "value = '1', ";
 			$query .= "date_time = '" . date('Y-m-d H:i:s') . "', ";
-			$query .= "ip = '$ip'";
+			$query .= "ip = '$ip', ";
+            $query .= "user_id = '$user_id'";
 		}
 	} else {
 		if($has_already_voted) {
@@ -78,12 +81,14 @@ if($can_vote) {
 			$query .= "date_time = '" . date('Y-m-d H:i:s') . "' ";
 			$query .= "WHERE post_id = '" . $post_id . "' AND ";
 			$query .= "ip = '$ip'";
+            $query .= "user_id = '$user_id'";
 		} else {
 			$query = "INSERT INTO {$wpdb->prefix}wti_like_post SET ";
 			$query .= "post_id = '" . $post_id . "', ";
 			$query .= "value = '-1', ";
 			$query .= "date_time = '" . date('Y-m-d H:i:s') . "', ";
-			$query .= "ip = '$ip'";
+			$query .= "ip = '$ip', ";
+            $query .= "user_id = '$user_id'";
 		}
 	}
 	//echo $query;
@@ -106,4 +111,4 @@ $wti_unlike_count = GetWtiUnlikeCount($post_id);
 
 $result = array("msg" => $msg, "error" => $error, "like" => $wti_like_count, "unlike" => $wti_unlike_count);
 
-echo json_encode($result);
\ No newline at end of file
+echo json_encode($result);
diff --git a/wti-like-post/wti_like_class.php b/wti-like-post/wti_like_class.php
index 468e33f..ad41cfd 100644
--- a/wti-like-post/wti_like_class.php
+++ b/wti-like-post/wti_like_class.php
@@ -252,4 +252,4 @@ class RecentlyLikedPosts
 }
 
 $RecentlyLikedPosts = new RecentlyLikedPosts();
-?>
\ No newline at end of file
+?>
diff --git a/wti-like-post/wti_like_post.php b/wti-like-post/wti_like_post.php
index 8be513a..70d41fc 100644
--- a/wti-like-post/wti_like_post.php
+++ b/wti-like-post/wti_like_post.php
@@ -360,6 +360,7 @@ function WtiLikePostAdminContent() {
 							<td>	
 								<input type="radio" name="wti_like_post_position" id="position_top" value="top" <?php if(('top' == get_option('wti_like_post_position')) || ('' == get_option('wti_like_post_position'))) { echo 'checked'; } ?> /> <?php echo __('Top of Content', 'wti-like-post'); ?>
 								<input type="radio" name="wti_like_post_position" id="position_bottom" value="bottom" <?php if('bottom' == get_option('wti_like_post_position')) { echo 'checked'; } ?> /> <?php echo __('Bottom of Content', 'wti-like-post'); ?>
+								<input type="radio" name="wti_like_post_position" id="position_none" value="none" <?php if('none' == get_option('wti_like_post_position')) { echo 'checked'; } ?> /> <?php echo __('None', 'wti-like-post'); ?>
 								<span class="description"><?php _e('Select the position where you want to show the like options.', 'wti-like-post')?></span>
 							</td>
 						</tr>			
@@ -600,11 +601,27 @@ function WtiLikePostAdminContent() {
 #### WIDGET ####
 require_once ABSPATH . 'wp-content/plugins/wti-like-post/wti_like_class.php';
 
+function have_avatar($user_id){
+    global $wpdb;
+
+    $results = $wpdb->get_results("
+      SELECT user_id, meta_value
+      FROM wp_usermeta
+      WHERE meta_key = 'simple_local_avatar'
+      AND user_id = " . $user_id
+    );
+
+    return $results;
+
+}
 #### FRONT-END VIEW ####
-function GetWtiLikePost($arg = null) {
-     global $wpdb;
-     $post_id = get_the_ID();
-     $wti_like_post = "";
+function GetWtiLikePostButton($arg = null){
+global $wpdb;
+$post_id = get_the_ID();
+$wti_like_post = "";
+
+
+  
 	
      //get the posts ids where we do not need to show like functionality
      $allowed_posts = explode(",", get_option('wti_like_post_allowed_posts'));
@@ -638,7 +655,7 @@ function GetWtiLikePost($arg = null) {
 	
 	//if excluded category, then dont show the like/dislike option
 	if($excluded) {
-		return;
+			return;
 	}
 	
 	//check for title text. if empty then have the default value
@@ -653,38 +670,149 @@ function GetWtiLikePost($arg = null) {
 	
 	//checking for excluded posts
      if(!in_array($post_id, $excluded_posts)) {		
-		$like_count = GetWtiLikeCount($post_id);
-		$unlike_count = GetWtiUnlikeCount($post_id);
-		$msg = GetWtiVotedMessage($post_id);
+//		$like_count = GetWtiLikeCount($post_id);
+        $like_count = '';
+//		$unlike_count = GetWtiUnlikeCount($post_id);
+        $unlike_count = '';
+        $cur_user = wp_get_current_user();
+		$msg = GetWtiVotedMessage($post_id,$cur_user->ID);
 		$alignment = ("left" == get_option('wti_like_post_alignment')) ? 'left' : 'right';
 		$show_dislike = get_option('wti_like_post_show_dislike');
 		$style = (get_option('wti_like_post_voting_style') == "") ? 'style1' : get_option('wti_like_post_voting_style');
 		
 		$wti_like_post .= "<div id='watch_action'>";
 		$wti_like_post .= "<div id='watch_position' style='float:".$alignment."; '>";
-		$wti_like_post .= "<div id='action_like' >".
-							"<span class='like-".$post_id." like'><img title='".__($title_text_like, 'wti-like-post')."' id='like-".$post_id."' rel='like' class='lbg-$style jlk' src='".WP_PLUGIN_URL."/wti-like-post/images/pixel.gif'></span>".
+		$wti_like_post .= "<div id='action_like' >";
+       if(is_null($msg)){
+            $wti_like_post .=" <span class='wti_like_message'>Bu bildiri işinize yaradıysa, tutmaya ne dersiniz? Yazıyı tutmak için yıldıza tıklayınız.</span><span class='like-".$post_id." like'><img title='".__($title_text_like, 'wti-like-post')."' id='like-".$post_id."' rel='like' class='xlbg-$style jlk' width='50' src='" . WP_PLUGIN_URL . "/wti-like-post/images/byildiz.png'></span>";
+        }
+       else {
+            $wti_like_post .="<span class='like-".$post_id." like'><img title='".__($title_text_like, 'wti-like-post')."' id='like-".$post_id."' rel='like' class='xlbg-$style jlk' width='50' src='" . WP_PLUGIN_URL . "/wti-like-post/images/dyildiz.png'></span>";
+        }
+            
 							/*"<span class='like-".$post_id." like'><img title='".__($title_text_like, 'wti-like-post')."' id='like-".$post_id."' rel='like' class='jlk' src='".WP_PLUGIN_URL."/wti-like-post/images/thumb_up_".$style.".png'></span>".*/
-							"<span id='lc-".$post_id."' class='lc'>".$like_count."</span>".
-					   "</div>";
+//							"<span id='lc-".$post_id."' class='lc'>".$like_count."</span>".
+        $wti_like_post .= "</div>";
 		
 		if($show_dislike) {
 			$wti_like_post .= "<div id='action_unlike' >".
-								"<span class='unlike-".$post_id." unlike'><img title='".__($title_text_unlike, 'wti-like-post')."' id='unlike-".$post_id."' rel='unlike' class='unlbg-$style jlk' src='".WP_PLUGIN_URL."/wti-like-post/images/pixel.gif'></span>".
+								"Bu bildiri işinize yaradıysa, tutmaya ne dersiniz? Yazıyı tutmak için yıldıza tıklayınız <span class='unlike-".$post_id." unlike'><img title='".__($title_text_unlike, 'wti-like-post')."' id='unlike-".$post_id."' rel='unlike' class='unlbg-$style jlk' src='".WP_PLUGIN_URL."/wti-like-post/images/pixel.gif'></span>".
 								/*"<span class='unlike-".$post_id." unlike'><img title='".__($title_text_unlike, 'wti-like-post')."' id='unlike-".$post_id."' rel='unlike' class='jlk' src='".WP_PLUGIN_URL."/wti-like-post/images/thumb_down_".$style.".png'></span>".*/
-								"<span id='unlc-".$post_id."' class='unlc'>".$unlike_count."</span>".
+//								"<span id='unlc-".$post_id."' class='unlc'>".$unlike_count."</span>".
 						   "</div> ";
 		}
 		
 		$wti_like_post .= "</div> ";
           $wti_like_post .= "<div id='status-".$post_id."' class='status' style='float:".$alignment."; '>&nbsp;&nbsp;" . $msg . "</div>";
-		$wti_like_post .= "</div><div id='clear'></div>";
+		$wti_like_post .= "</div><div id='clear'></div>"; 
+
+     } 
+     
+    
+		return $wti_like_post;
+     
+}
+function GetWtiLikePost($arg = null){
+global $wpdb;
+$post_id = get_the_ID();
+$likeds = '';
+$liked_users = $wpdb->get_results("
+                      SELECT post.user_id 
+                      FROM wp_wti_like_post post
+                      LEFT JOIN wp_users user
+                        ON post.user_id = user.ID
+                       WHERE post.post_id =" . $post_id
+                      );
+$k = 0;
+$i = 0;
+$count=count($liked_users);
+foreach($liked_users as $liked){
+  $k++;
+  $user_infos = get_userdata($liked->user_id);  
+  $user_name = str_replace(' ', '-', $user_infos->data->user_login);
+
+    $avatar_info = have_avatar($liked->user_id);
+    if( empty($avatar_info)){
+    	$likeds .= '<a style="float: left;" class="wti_avatar" href="/uye/' . $user_name . '">'.get_avatar($liked->user_id, 50).'</a>';
+    }
+    else{
+        $unserialized = unserialize($avatar_info[0]->meta_value);
+        $avatar = isset($unserialized[80]) ? $unserialized[80]  : $unserialized[96];
+
+     	$likeds .= '<a style="float: left;" class="wti_avatar" href="/uye/'.$user_name.'"><img width="50" style="margin:0 10px;" src="' . $avatar . '" /></a>';
+       } 
+    /*if($k==5){
+        $likeds .= '<div style="clear: both;"></div><br />';
+        $k = 0;
+    }*/
+}
+
+    if ( ! is_user_logged_in())
+        return $likeds;
+
+     
+	
+     //get the posts ids where we do not need to show like functionality
+     $allowed_posts = explode(",", get_option('wti_like_post_allowed_posts'));
+     $excluded_posts = explode(",", get_option('wti_like_post_excluded_posts'));
+	$excluded_categories = get_option('wti_like_post_excluded_categories');
+	$excluded_sections = get_option('wti_like_post_excluded_sections');
+	
+	if(empty($excluded_categories)) {
+		$excluded_categories = array();
+	}
+	
+	if(empty($excluded_sections)) {
+		$excluded_sections = array();
+	}
+     
+	$title_text = get_option('wti_like_post_title_text');
+	$category = get_the_category();
+	$excluded = false;
+	
+	//checking for excluded section. if yes, then dont show the like/dislike option
+	if((in_array('home', $excluded_sections) && is_home()) || (in_array('archive', $excluded_sections) && is_archive())) {
+		return;
+	}
+	
+	//checking for excluded categories
+	foreach($category as $cat) {
+		if(in_array($cat->cat_ID, $excluded_categories) && !in_array($post_id, $allowed_posts)) {
+			$excluded = true;
+		}
+	}
+	
+	//if excluded category, then dont show the like/dislike option
+	if($excluded) {
+			return;
+	}
+	
+	//check for title text. if empty then have the default value
+     if(empty($title_text)) {
+		$title_text_like = __('Like', 'wti-like-post');
+		$title_text_unlike = __('Unlike', 'wti-like-post');
+     } else {
+		$title_text = explode('/', get_option('wti_like_post_title_text'));
+		$title_text_like = $title_text[0];
+		$title_text_unlike = $title_text[1];
      }
+	
+	//checking for excluded posts
+     if(!in_array($post_id, $excluded_posts)) {		
+//		$like_count = GetWtiLikeCount($post_id);
+        $like_count = '';
+//		$unlike_count = GetWtiUnlikeCount($post_id);
+        $unlike_count = '';
+        $cur_user = wp_get_current_user();
+		$msg = GetWtiVotedMessage($post_id,$cur_user->ID);
+
+     } 
      
      if ($arg == 'put') {
-		return $wti_like_post;
+		return $wti_like_post . $likeds;
+		//return $likeds;
      } else {
-		echo $wti_like_post;
+		return $wti_like_post;
      }
 }
 
@@ -703,6 +831,8 @@ function PutWtiLikePost($content) {
 			$content = $wti_like_post_content . $content;
 		} elseif ($wti_like_post_position == 'bottom') {
 			$content = $content . $wti_like_post_content;
+		} elseif ($wti_like_post_position == 'none') {
+			$content = $content;
 		} else {
 			$content = $wti_like_post_content . $content . $wti_like_post_content;
 		}
@@ -711,7 +841,7 @@ function PutWtiLikePost($content) {
      return $content;
 }
 
-add_filter('the_content', 'PutWtiLikePost');
+add_filter('the_content',"PutWtiLikePost");
 
 function GetWtiLikeCount($post_id) {
      global $wpdb;
@@ -748,7 +878,7 @@ function GetWtiUnlikeCount($post_id) {
      return $wti_unlike_count;
 }
 
-function GetWtiVotedMessage($post_id, $ip = null) {
+function GetWtiVotedMessage($post_id, $user_id = null) {
      global $wpdb;
 	
      if(null == $ip)
@@ -756,7 +886,7 @@ function GetWtiVotedMessage($post_id, $ip = null) {
 		$ip = $_SERVER['REMOTE_ADDR'];
      }
      
-     $wti_has_voted = $wpdb->get_var("SELECT COUNT(id) AS has_voted FROM {$wpdb->prefix}wti_like_post WHERE post_id = '$post_id' AND ip = '$ip'");
+     $wti_has_voted = $wpdb->get_var("SELECT COUNT(id) AS has_voted FROM {$wpdb->prefix}wti_like_post WHERE post_id = '$post_id' AND user_id = '$user_id'");
      
      if($wti_has_voted > 0) {
 		$wti_voted_message = get_option('wti_like_post_voted_message');
@@ -765,7 +895,7 @@ function GetWtiVotedMessage($post_id, $ip = null) {
      return $wti_voted_message;
 }
 
-function HasWtiAlreadyVoted($post_id, $ip = null) {
+function HasWtiAlreadyVoted($post_id, $user_id = null) {
      global $wpdb;
      
      if(null == $ip)
@@ -773,7 +903,7 @@ function HasWtiAlreadyVoted($post_id, $ip = null) {
 		$ip = $_SERVER['REMOTE_ADDR'];
      }
 	
-     $wti_has_voted = $wpdb->get_var("SELECT COUNT(id) AS has_voted FROM {$wpdb->prefix}wti_like_post WHERE post_id = '$post_id' AND ip = '$ip'");
+     $wti_has_voted = $wpdb->get_var("SELECT COUNT(id) AS has_voted FROM {$wpdb->prefix}wti_like_post WHERE post_id = '$post_id' AND user_id = '$user_id'");
      
      return $wti_has_voted;
 }
@@ -1097,4 +1227,4 @@ function wti_like_post_save_data($post_id) {
 		}
 	}
 }
-?>
\ No newline at end of file
+?>
