diff --git a/wti-like-post/wti_like_post.php b/wti-like-post/wti_like_post.php
index 30c6c9f..85a400e 100755
--- a/wti-like-post/wti_like_post.php
+++ b/wti-like-post/wti_like_post.php
@@ -684,7 +684,7 @@ $wti_like_post = "";
 		$wti_like_post .= "<div id='watch_position' style='float:".$alignment."; '>";
 		$wti_like_post .= "<div id='action_like' >";
        if(is_null($msg)){
-            $wti_like_post .=" <span class='wti_like_message'>Bu bildiri işinize yaradıysa, tutmaya ne dersiniz? Yazıyı tutmak için yıldıza tıklayınız</span><span class='like-".$post_id." like'><img title='".__($title_text_like, 'wti-like-post')."' id='like-".$post_id."' rel='like' class='xlbg-$style jlk' width='50' src='/wp-content/themes/byildiz.png'></span>";
+            $wti_like_post .=" <span class='wti_like_message'>Bu bildiri işinize yaradıysa, tutmaya ne dersiniz? Yazıyı tutmak için yıldıza tıklayınız.</span><span class='like-".$post_id." like'><img title='".__($title_text_like, 'wti-like-post')."' id='like-".$post_id."' rel='like' class='xlbg-$style jlk' width='50' src='/wp-content/themes/byildiz.png'></span>";
         }
        else {
             $wti_like_post .="<span class='like-".$post_id." like'><img title='".__($title_text_like, 'wti-like-post')."' id='like-".$post_id."' rel='like' class='xlbg-$style jlk' width='50' src='/wp-content/themes/dyildiz.png'></span>";

diff --git a/wti-like-post/wti_like_post.php b/wti-like-post/wti_like_post.php
index 85a400e..e15d993 100755
--- a/wti-like-post/wti_like_post.php
+++ b/wti-like-post/wti_like_post.php
@@ -733,13 +733,13 @@ foreach($liked_users as $liked){
 
     $avatar_info = have_avatar($liked->user_id);
     if( empty($avatar_info)){
-    	$likeds .= '<a style="float: left;" href="/uye/' . $user_name . '">' . get_avatar($liked->user_id, 50) . ' </a>';
+    	$likeds .= '<a style="float: left;" class="wti_avatar" href="/uye/' . $user_name . '">' . get_avatar($liked->user_id, 50) . ' </a>';
     }
     else{
         $unserialized = unserialize($avatar_info[0]->meta_value);
         $avatar = isset($unserialized[80]) ? $unserialized[80]  : $unserialized[96];
 
-     	$likeds .= '<a style="float: left;" href="/uye/'.$user_name.'"><img width="50" style="margin:0 10px;" src="' . $avatar . '" /></a>';
+     	$likeds .= '<a style="float: left;" class="wti_avatar" href="/uye/'.$user_name.'"><img width="50" style="margin:0 10px;" src="' . $avatar . '" /></a>';
        } 
     /*if($k==5){
         $likeds .= '<div style="clear: both;"></div><br />';

diff --git a/wti-like-post/wti_like_post.php b/wti-like-post/wti_like_post.php
index e15d993..600534c 100755
--- a/wti-like-post/wti_like_post.php
+++ b/wti-like-post/wti_like_post.php
@@ -733,7 +733,7 @@ foreach($liked_users as $liked){
 
     $avatar_info = have_avatar($liked->user_id);
     if( empty($avatar_info)){
-    	$likeds .= '<a style="float: left;" class="wti_avatar" href="/uye/' . $user_name . '">' . get_avatar($liked->user_id, 50) . ' </a>';
+    	$likeds .= '<a style="float: left;" class="wti_avatar" href="/uye/' . $user_name . '">'.get_avatar($liked->user_id, 50).'</a>';
     }
     else{
         $unserialized = unserialize($avatar_info[0]->meta_value);
